docs/reference/core.md

```diff
@@ -7,7 +7,11 @@ functions.  Use at your own risk.
 ## Multipart Upload
 
 ::: synapseclient.core.upload.multipart_upload
-::: synapseclient.core.upload.multipart_upload._multipart_upload
+
```

Comment:
I removed the explicit call to `::: synapseclient.core.upload.multipart_upload._multipart_upload` because it looked confusing in the documentation, and I wanted to have a conversation about it later. From a quick scan of the documentation, this looks to be the only hidden function that's explicitly documented. I see why that is. The returns of the other functions are calls to `_multipart_upload`, and so references to it are made in the Multipart section of Core. But this goes back to a previous conversation of whether or not we should expose the documentation of a hidden function and vice versa. To me if something is hidden with `_`, it's best not to add it to the documentation because it will be hidden from the user unless it's explicitly called, so it's best to avoid possible confusion. But I'm up for iterating on this before a final review.

Reply:
One thing we can explicitly state is that people should use functions within `core` at their own risk.   There's not really anything "hidden" within Python packages, but the functions within `core` are for developers only and aren't part of the "public" Python API.

Reply:
What I mean by hidden is that the functions led by `_` are denoted to be for internal use only, and won't be available unless explicitly called, so it's not functionality that's immediately available to users (a user would have to look inside the code to know it's there).

![image](https://github.com/Sage-Bionetworks/synapsePythonClient/assets/32107699/11728140-59d2-49e9-8cf3-a4e2ae1a478e)

But based on what you're saying, it sounds like we want to document everything in `core`, both internal and external functions since `core` is for users that are more advanced to begin with. If that's the case, I would add and update the docstring for `_multipart_upload` and also expose `retry.py`'s [_get_message](https://github.com/Sage-Bionetworks/synapsePythonClient/blob/e80fa0d7816e6c97bb60c55f35d72a57d742d6c0/synapseclient/core/retry.py#L171)

Reply:
I think all functions in `core` and this package should have docstrings even if they aren't surfaced on the online python documentation site.

I don't think we actually need all of `core` to be available in the online python documentation site, but we should document that the functions in `core` are subject to change at any time and for python client users to use at their own risk.